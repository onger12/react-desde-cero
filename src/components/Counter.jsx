import React from "react";
import PropTypes from 'prop-types'
import { useState } from "react";

const Counter = ({ value }) => {
    const [counter, setCounter] = useState( value )

    const handleAdd = () => setCounter( counter + 1 );
    const handleSubstract = () => setCounter( counter - 1 );
    const handleReset = () => setCounter( value );

    return (
        <>
            <h1>Counter App</h1>
            <h2> { counter } </h2>
            <button aria-label="btn-+1" onClick={ handleAdd }> +1 </button>
            <button aria-label="btn--1" onClick={ handleSubstract }> -1 </button>
            <button aria-label="btn-reset" onClick={ handleReset }> reset </button>
        </>
    )
}


Counter.propTypes = {
    value: PropTypes.number.isRequired
}

Counter.defaultProps = {
    value: 0
}

export default Counter