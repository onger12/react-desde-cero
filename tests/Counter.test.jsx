import { fireEvent, render, screen } from '@testing-library/react'
import Counter from '../src/components/Counter'

describe('Pruebas en <Counter />', () => {
  
    const value = 10;

    test('debe hacer match con el snapshot', () => {
        
        const { container } = render( <Counter value={ value } /> );
        
        expect( container ).toMatchSnapshot();

    })
    
    test('debe renderizar el valor por defecto pasado por props', () => {
        
        render( <Counter value={ value }/> );
        
        expect( screen.getByText(value) ).toBeTruthy();

    })

    test('debe incrementar el contador cuando se presione el boton +1', () => {
        
        const { container, debug } = render( <Counter value={ value } /> );
        
        fireEvent.click( screen.getByRole('button', { name: "btn-+1" }) );
        
        expect( screen.findByText( value + 1 ) ).toBeTruthy();
    })
    test('debe decrementar el contador cuando se presione el boton -1', () => {
        
        const { container, debug } = render( <Counter value={ value } /> );
        
        fireEvent.click( screen.getByRole('button', { name: "btn--1" }) );
        
        expect( screen.findByText( value - 1 ) ).toBeTruthy();
    })
    test('debe resetear el contador cuando se presione el boton reset', () => {
        
        const { container, debug } = render( <Counter value={ value } /> );

        fireEvent.click( screen.getByRole('button', { name: "btn-+1" }) );
        fireEvent.click( screen.getByRole('button', { name: "btn-+1" }) );
        fireEvent.click( screen.getByRole('button', { name: "btn-+1" }) );
        fireEvent.click( screen.getByRole('button', { name: "btn-reset" }) );

        expect( screen.findByText( value ) ).toBeTruthy();
    })

})