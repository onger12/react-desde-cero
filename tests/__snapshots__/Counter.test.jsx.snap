// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Counter /> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    aria-label="btn-+1"
  >
     +1 
  </button>
  <button
    aria-label="btn--1"
  >
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     reset 
  </button>
</div>
`;
